test/integration/index.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const {test} = require('tap')
+const config = require('../../config.js')
+const nock = require('nock')
+const pkg = require('../../package.json')
+const URL = 'https://ingest.pipeline.com'
+const KEY = 'abc123'
+process.env.PIPELINE_KEY = KEY
+process.env.PIPELINE_URL = URL
```

Comment:
A few things here:

- Dom and Matt already mentioned this, but you only need to defined the props on `process.env` and then let the config loader do its thing. I'm not sure what issue you were seeing, but you can apply the diff below and verify that it works (confirmed myself)
- Here and in the `index.js` at the root, we're missing a call to `config.validateEnvVars()` which enforces the required config. See the diff below. It's critical that we include that in the entrypoint, since it will ensure the user gets clear error messages when environment variables are missing.
- nit: I don't think it's causing any problems, but worth noting here you are shadowing the `URL` [global](https://developer.mozilla.org/en-US/docs/Web/API/URL).

```diff
diff --git a/test/integration/index.js b/test/integration/index.js
index 75c1ba7..dbe10a7 100644
--- a/test/integration/index.js
+++ b/test/integration/index.js
@@ -1,13 +1,14 @@
 'use strict'
 
+process.env.PIPELINE_URL = 'https://ingest.pipeline.com'
+process.env.PIPELINE_KEY = 'abc123'
+
 const {test} = require('tap')
-const config = require('../../config.js')
 const nock = require('nock')
+const config = require('../../config.js')
 const pkg = require('../../package.json')
-const URL = 'https://ingest.pipeline.com'
-const KEY = 'abc123'
-process.env.PIPELINE_KEY = KEY
-process.env.PIPELINE_URL = URL
+
+config.validateEnvVars()
 
 // Internal Modules
 const index = require('../../index.js')
@@ -55,17 +56,14 @@ const expectedReqBody = [
 ]
 
 test('integration tests', async (t) => {
-  config.set('pipeline-key', KEY)
-  config.set('pipeline-url', URL)
-
   t.test('successful send message', async (t) => {
 
     nock(
-      URL,
+      config.get('pipeline-url'),
       {
         reqheaders:
         {
-          'authorization': KEY
+          'authorization': config.get('pipeline-key')
         , 'user-agent': `${pkg.name}/${pkg.version}`
         }
       }
@@ -79,8 +77,7 @@ test('integration tests', async (t) => {
   })
 
   t.test('failure send message', async (t) => {
-
-    nock(URL).persist().post('/').reply(500)
+    nock(config.get('pipeline-url')).persist().post('/').reply(500)
     await index.handler(rawEvent, {}, (error) => {
       t.type(error, 'HTTPError')
     })
```

Reply:
Thanks.  I think I just didn't have it high enough when I tried this.  So config.js was loading before the process.env(s) were set.